packages/core-api/src/app/App.tsx

```diff
@@ -169,8 +169,8 @@ export class PrivateAppImpl implements BackstageApp {
     return this.plugins;
   }
 
-  getSystemIcon(key: IconKey): IconComponent {
-    return this.icons[key];
+  getSystemIcon(key: IconKey | undefined): IconComponent {
+    return this.icons[key ?? ''];
```

Comment:
Isn't this a bit of an odd interface change? Won't this typically always return undefined, which would break the typescript contract.

If anything, this one should instead be changed to return `IconComponent | undefined` or return a fallback icon - but to keep the `key` argument as it was. Does that make sense?

Reply:
Right, I originally tried exactly what your suggesting, but this method is used here https://github.com/backstage/backstage/blob/master/packages/core-api/src/icons/icons.tsx#L36 which then could result in an undefined JSX node. It probably would be a very outlier case, but this could blow up if the integrator somehow deleted the system icon mapping?

Reply:
I got this to work, but only if I use `null` vs `undefined`. It appears that since this is actually a `ComponentType` it can't be undefined, from how I'm reading the typing errors.

Reply:
I think returning undefined here was the right choice. This is just a regular API method call. It happens to return a component but that's actually not important; it's not supposed to return something that is always directly placeable in JSX, and it's not supposed to be callable with weird arguments. :)

I suggest, leave this as `getSystemIcon(key: IconKey): IconComponent | undefined`, because that conveys that (1) you are supposed to give it an icon key - which includes any string - and (2) you have to be prepared for that what you asked for may not exist.

Otherwise, change it so that it returns `IconComponent` always but it may be a fallback instead of the one you thought you asked for.

so this one would either return just plain `this.icons[key]` and let that be undefined, or `this.icons[key] || SomeFallbackIcon`.

Then you update the call sites of this method. That means adding `if`-statements and checks, sure.

Reply:
But after that, I think this is great and good to go

Reply:
I could also update or add a helper to do this:

```ts
  const getIcon = (
    key: IconKey | undefined,
    fallbackType: IconComponent,
  ): IconComponent =>
    key ? app.getSystemIcon(key) ?? fallbackType : fallbackType;
```

Since I'm ending up doing the following: 

```ts
key ? app.getSystemIcon(key) ?? LanguageIcon : LanguageIcon
```

Reply:
Sure, or sneakily `app.getSystemIcon(key || '') || MyFallbackIcon` even though that is a hack in comparison :)

Reply:
Think I'll leave it as is for now if that's ok.